pkg/controllers/ttl-controller/manager.go

```diff
@@ -0,0 +1,186 @@
+	"k8s.io/client-go/metadata/metadatainformer"
+	"k8s.io/client-go/tools/cache"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	authorizationv1client "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+type stopFunc = context.CancelFunc
+
+type manager struct {
+	metadataClient  metadata.Interface
+	discoveryClient discovery.DiscoveryInterface
+	checker         checker.AuthChecker
+	resController   map[schema.GroupVersionResource]stopFunc
+}
+
+func NewManager(metadataInterface metadata.Interface, discoveryInterface discovery.DiscoveryInterface,  authorizationInterface authorizationv1client.AuthorizationV1Interface) (*manager, error) {
+	// client
+	metadataClient := metadataInterface
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// discovery client
+	discoveryClient:= discoveryInterface
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	authClient := authorizationInterface
+
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// checker
+
+	selfChecker := checker.NewSelfChecker(authClient.SelfSubjectAccessReviews())
+
+	resController := make(map[schema.GroupVersionResource]stopFunc)
+
+	return &manager{
+		metadataClient:  metadataClient,
+		discoveryClient: discoveryClient,
+		checker:         selfChecker,
+		resController:   resController,
+	}, nil
+}
+
+func (m *manager) Run(ctx context.Context) error {
+	defer func() {
+		// Stop all informers and wait for them to finish
+		for gvr := range m.resController {
+			if err := m.stop(ctx, gvr); err != nil {
+				log.Println("Error stopping informer:", err)
+			}
+		}
+	}()
+
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			if err := m.reconcile(ctx); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (m *manager) getDesiredState() (sets.Set[schema.GroupVersionResource], error) {
+	// Get the list of resources currently present in the cluster
+	newresources, err := discoverResources(m.discoveryClient)
+	if err != nil {
+		return nil, err
+	}
+	validResources := m.filterPermissionsResource(newresources)
+	return sets.New(validResources...), nil
+}
+
+func (m *manager) getObservedState() (sets.Set[schema.GroupVersionResource], error) {
+	observedState := sets.New[schema.GroupVersionResource]()
+	for resource := range m.resController {
+		observedState.Insert(resource)
+	}
+	return observedState, nil
+}
+
+func (m *manager) stop(ctx context.Context, gvr schema.GroupVersionResource) error {
+	if stopFunc, ok := m.resController[gvr]; ok {
+		delete(m.resController, gvr)
+		func() {
+			// defer log.Println("stopped", gvr)
+			log.Println("stopping...", gvr)
+			stopFunc()
+		}()
+	}
+	return nil
+}
+
+func (m *manager) start(ctx context.Context, gvr schema.GroupVersionResource) error {
+	indexers := cache.Indexers{
+		cache.NamespaceIndex: cache.MetaNamespaceIndexFunc,
+	}
+	options := func(options *metav1.ListOptions) {
+		options.LabelSelector = "kyverno.io/ttl"
```

Comment:
Better to extract `kyverno.io/ttl` to a constant, we could re-use it everywhere.